### PR TITLE
feat(explored): add the HostByPubkey API

### PR DESCRIPTION
**Pull request by telestrial**

### CODE REVIEW

**Comment by telestrial** [comment]
* **#819** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/819?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#818** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/818?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/818?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: libs/explored-types/src/types.ts [hunk 1]
```diff
@@ -104,14 +105,148 @@ export type TxpoolTransactions = {
   v2transactions: V2Transaction[]
 }
 
+/**
+ * HostSettings are the settings and prices used when interacting with a host.
+ */
+export type HostSettings = {
```
Comment (alexfreska) [inline libs/explored-types/src/types.ts:111]: I think we have the same types already in `renterd-types`: https://github.com/SiaFoundation/web/blob/704d5a629f011abc89ba3c6b61004d01a46af22b/libs/renterd-types/src/types.ts#L23-L84

Could you move them into `types` and import then into `renterd-types` and `explored-types`.

**Review by alexfreska — COMMENTED** [review]
Looks good, but I think we can dedupe the host types.

File: libs/explored-types/src/types.ts (hunk 1 above)
Comment (telestrial) [reply libs/explored-types/src/types.ts]: Sure thing. Handled in a down stack PR.

**Review by alexfreska — APPROVED** [review]

**Comment by telestrial** [comment]
### Merge activity

* **Nov 21, 12:47 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/SiaFoundation/web/818).

**Pull request merged**